### PR TITLE
FE: Node Status updates

### PR DESCRIPTION
**Pull request by Syed-Bilal-Haider-Dataq**
## Description
Update the inventoryStore with a Map of node ID to boolean status.

In inventoryStore.init() in addition to making a call to buildNetworkInventory , also make a call to Node Status Service, allNodeStatus. 

Ideally, make parallel calls to these, refactor to use Promise.all, etc.

When that returns, fill the inventoryStore.nodeStatusMap.

Then on the Node Status page, use that value to display the node’s Status in both the table and the card.

Ideally, the topNNode call in dashboardStore should save any node status it finds into the inventoryStore.nodeStatusMap.

Eventually we may need to set up polling of the node status so that they update in semi-realtime.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2710

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Make discussed changes and let me know

File: ui/src/components/Inventory/InventoryTabContent.vue [hunk 1]
```diff
@@ -97,14 +99,18 @@ const openModalForDeletingTags = (node: NewInventoryNode) => {
   tagStore.openModal()
 }
 
-const metricsAsTextBadges = (metrics?: RawMetric) => {
+const metricsAsTextBadges = (metrics?: RawMetric, nodeId?: string) => {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabContent.vue:102]: this function is common in `Table` and `Card` component. Move it to `inventory.utils.ts`

File: ui/src/components/Inventory/InventoryTabContent.vue [hunk 2]
```diff
@@ -97,14 +99,18 @@ const openModalForDeletingTags = (node: NewInventoryNode) => {
...
+  if (!nodeId) {
+    return []
+  }
   const badges = []
+  const label = nodeStatus.value.get(nodeId)
 
-  if (metrics?.value?.[1]) {
-    badges.push({ type: BadgeTypes.success, label: metrics.value?.[1] + 'ms' })
-    badges.push({ type: BadgeTypes.success, label: 'Up' })
+  if (label?.trim().toLocaleLowerCase() === 'up') {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabContent.vue:109]: change `.toLocaleLowerCase()` to `.toLowerCase()` if you are not passing `Locale` here

File: ui/src/components/Inventory/InventoryTabTable.vue [hunk 3]
```diff
@@ -139,12 +140,15 @@ const onPageSizeChanged = (p: number) => {
   stopSpinner()
 }
 
-const metricsAsTextBadges = (metrics?: RawMetric) => {
+const metricsAsTextBadges = (metrics?: RawMetric, nodeId?: string | undefined) => {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabTable.vue:143]: Move this function to `inventory.utils.ts`

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Just small fixes. Let me know when you are done.

**Review by Shahbaz-dataq — APPROVED** [review]
LGTM

File: ui/src/store/Views/inventoryStore.ts [hunk 4]
```diff
@@ -39,32 +39,47 @@ export const useInventoryStore = defineStore('inventoryStore', {
...
+      try {
+        const page = this.inventoryNodesPagination.page > 0 ? this.inventoryNodesPagination.page - 1 : 0
+        const pagination = {
+          ...this.inventoryNodesPagination,
+          page
         }
-        this.totalInventoryNodes = data.searchNodes.totalNodes || 0
-        if (this.inventoryNodesPagination.total !== this.totalInventoryNodes) {
-          this.inventoryNodesPagination.total = this.totalInventoryNodes
+        const [data, allNodesStatus] = await Promise.all([
```
Comment (synqotik) [inline ui/src/store/Views/inventoryStore.ts:55]: Good use of `Promise.all` !

**Review by synqotik — APPROVED** [review]
LGTM

**Pull request merged**